FlowCrypt/Functionality/Services/Compose Message Service/ComposeMessageService.swift

```diff
@@ -101,48 +103,49 @@ final class ComposeMessageService {
                 )
             }
 
-        return getPubKeys(for: recipients)
-            .mapError { ComposeMessageError.validationError($0) }
-            .map { allRecipientPubs in
-                let replyToMimeMsg = input.replyToMime
-                    .flatMap { String(data: $0, encoding: .utf8) }
-
-                return SendableMsg(
-                    text: text,
-                    to: recipients.map(\.email),
-                    cc: [],
-                    bcc: [],
-                    from: email,
-                    subject: subject,
-                    replyToMimeMsg: replyToMimeMsg,
-                    atts: sendableAttachments,
-                    pubKeys: allRecipientPubs + [myPubKey],
-                    signingPrv: signingPrv
-                )
-            }
+        let allRecipientPubs = try await getPubKeys(for: recipients)
+        let replyToMimeMsg = input.replyToMime
+            .flatMap { String(data: $0, encoding: .utf8) }
+        return SendableMsg(
+            text: text,
+            to: recipients.map(\.email),
+            cc: [],
+            bcc: [],
+            from: email,
+            subject: subject,
+            replyToMimeMsg: replyToMimeMsg,
+            atts: sendableAttachments,
+            pubKeys: [myPubKey] + allRecipientPubs,
+            signingPrv: signingPrv
+        )
```

Comment:
includeAttachments boolean is ignored but should be used, as far as drafts don't need to save attachments.

Reply:
we could create a separate issue for this since this PR is already huge

Reply:
Drafts indeed shouldn't save attachments. Can file in separate issue.

Reply:
PR is huge indeed, but we should care about old functionality while refactoring new one.
Its really important, because otherwise PRs will produce more pain than effort.

I think PRs like that should be initially split into several ones ( lets try to do like that in future ). Review and ensure that everything that was refactored is not broken - its very hard in current PR.

Reply:
Yes - I'll fix that in this PR.

This particular PR could not be split easily. Core methods were made async to prevent a bug, and that made the whole app async because they use the core methods.

Reply:
But in general we always try to make refactoring PRs smaller whenever possible.

Reply:
It looks like the original code also ignored it, which is why Ivan proposed to do it in another PR. https://github.com/FlowCrypt/flowcrypt-ios/blob/41a5a6127f5bcdadcc892859ad10beca7cdc0513/FlowCrypt/Functionality/Services/Compose%20Message%20Service/ComposeMessageService.swift#L56 `includeAttachments` is unused.

Reply:
I fixed it in this PR